CMakeLists.txt

```diff
@@ -94,7 +94,7 @@ endif()
 
 # System dependencies
 find_package(dispatch CONFIG)
-if(NOT dispatch_FOUND)
+if(NOT dispatch_FOUND AND NOT CMAKE_SYSTEM_NAME STREQUAL WASI)
```

Comment:
If WASI doesn't have dispatch, can we wrap this whole section in an `if` block to avoid even trying `find_package(dispatch CONFIG)` if we know it won't (or shouldn't) exist?